### ISSUE TITLE
[Bug] Plugin breaks diff highlighting in reading view

### ISSUE THREAD

**Issue by cgarrard** [mugiwara85/CodeblockCustomizer#58]
Take this `diff` codeblock:

````text
```diff
@@ -1 +1,2 @@
-print('hello world')
+for i in range(10):
+    print('hello world')
```
````

It displays correctly in edit mode:

![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/4453816/61df80a2-62ac-436f-a38d-f71cfa857291)

But only the first added line is green when in reading mode:

![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/4453816/d04e0db1-13b3-4f9e-b043-df75c39bd735)

If I disable the plugin then it displays as expected in the reading view:

![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/4453816/8929f252-b6b7-4a9e-936a-85a48eab7193)

I'm using the Things theme, but I tried using the default theme and that didn't change anything. Disabling line numbers didn't help either.

Thank you for creating this plugin-- I really like it!

**Comment by mugiwara85** [comment]
I will look at it. Thanks for the sample code block

**Comment by ColinOppenheim** [comment]
I'm having the same problem:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/83012160-2c7d-471c-8e68-669d70fe63c4)
Here is the same content in reading view:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/b71ee77b-709c-4968-bd0c-4801b65d1317)

However disabling the plugin  and then re enabling it hides all the content T had modified in reading view. I have to go into edit mode. Cut all the content and repaste and then it shows up with the same issue as shown above; however, I decided to remove the hl:3 from the code block since it wasn't working so the code block looks like this now:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/2dc7f122-7314-4a30-9b74-53710f3fcda9)

But when I switch back to reading mode:
Voila!
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/d2f84400-818b-47e8-9d8d-c455e807a43c)

Yea i have no idea why. I don't have many plugins enabled just one or two and this is the only one that provides any level of visual enhancement. If it helps your troubleshooting i have included my current settings for the plugin below:
[data.json](https://github.com/mugiwara85/CodeblockCustomizer/files/13462412/data.json)

I'm using a Theme marked as (Legacy) called Tokyo Night... not sure if that has anything to do with it... Hope this information helps you figure it out.

**Comment by mugiwara85** [comment]
> I'm having the same problem: ![image](https://private-user-images.githubusercontent.com/23444760/285550512-83012160-2c7d-471c-8e68-669d70fe63c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDA4NjY1NTksIm5iZiI6MTcwMDg2NjI1OSwicGF0aCI6Ii8yMzQ0NDc2MC8yODU1NTA1MTItODMwMTIxNjAtMmM3ZC00NzFjLThlNjgtNjY5ZDcwZmU2M2M0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI0VDIyNTA1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxNmQ0MTE0MmUzMzMxYjA2NGRkNTQ0NTNjMGMzYzM2OWFmMzAyOTE5NmFlMjU3MGJiYzdlNWY4ZGY2NmRjMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZJrlfr70whEIgMThC_1BeF9CXl4tIWwsOXHwrexFpAw) Here is the same content in reading view: ![image](https://private-user-images.githubusercontent.com/23444760/285550801-b71ee77b-709c-4968-bd0c-4801b65d1317.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDA4NjY1NTksIm5iZiI6MTcwMDg2NjI1OSwicGF0aCI6Ii8yMzQ0NDc2MC8yODU1NTA4MDEtYjcxZWU3N2ItNzA5Yy00OTY4LWJkMGMtNDgwMWI2NWQxMzE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI0VDIyNTA1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiYjAzNGRiODkyNzI3YzczNjBkNGZhOGE0MjA3NjM1NzAwZjBjMmFmOTdlZDk0NDBhMDI0ZmM1ZjcxMjdkZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xrCKBCZlbL04JXFfOuHN3w43193aRWM-BVwC94luwis)
> 
> However disabling the plugin and then re enabling it hides all the content T had modified in reading view. I have to go into edit mode. Cut all the content and repaste and then it shows up with the same issue as shown above; however, I decided to remove the hl:3 from the code block since it wasn't working so the code block looks like this now: ![image](https://private-user-images.githubusercontent.com/23444760/285551388-2dc7f122-7314-4a30-9b74-53710f3fcda9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDA4NjY1NTksIm5iZiI6MTcwMDg2NjI1OSwicGF0aCI6Ii8yMzQ0NDc2MC8yODU1NTEzODgtMmRjN2YxMjItNzMxNC00YTMwLTliNzQtNTM3MTBmM2ZjZGE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI0VDIyNTA1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjZTY5MDkxMjQ5NmZlZmE0YzRiYjJiMGJiY2MzZGVlM2IxYTAxNDQ4NDM1MTQ2MzlkYTE0YWI4NWUzMDliZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._8OrATQhpPruVH8vJRbCBBOc2m91UrUoIrIRjuD2aOA)
> 
> But when I switch back to reading mode: Voila! ![image](https://private-user-images.githubusercontent.com/23444760/285551297-d2f84400-818b-47e8-9d8d-c455e807a43c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDA4NjY1NTksIm5iZiI6MTcwMDg2NjI1OSwicGF0aCI6Ii8yMzQ0NDc2MC8yODU1NTEyOTctZDJmODQ0MDAtODE4Yi00N2U4LTlkOGQtYzQ1NWU4MDdhNDNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI0VDIyNTA1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxM2Y4ZmFhNjgyNDA2MGE3MmU4ZjNkYTcyNjU1ODdjM2QwNzM2Nzk0MGM3OWQxMzhlMWE5MmM4YjhkZjMyZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FwGo1wSL1IEB494wliyXjSy_WxxHWNAAU9Iml1YJwmQ)
> 
> Yea i have no idea why. I don't have many plugins enabled just one or two and this is the only one that provides any level of visual enhancement. If it helps your troubleshooting i have included my current settings for the plugin below: [data.json](https://github.com/mugiwara85/CodeblockCustomizer/files/13462412/data.json)
> 
> I'm using a Theme marked as (Legacy) called Tokyo Night... not sure if that has anything to do with it... Hope this information helps you figure it out.

Yeah, the info definitely helps to debug. Can you please send the code block as well, so I can reproduce it exactly? This might actually be two problems. I will look at it, what could be the problem.

**Comment by ColinOppenheim** [comment]
These are two code blocks I attempted to highlight...

``` java
FXMLLoader loader = new FXMLLoader(getClass().getResource("/Views/NextScreen.fxml"));  
Parent root = loader.load();  
Stage stage = (Stage) ((Button) event.getSource()).getScene().getWindow();  
stage.setScene(new Scene(root));  
stage.show();
```
```java
FXMLLoader loader = new FXMLLoader(getClass().getResource("/Views/NextScreen.fxml"));  
Parent root = loader.load();  
Stage stage = (Stage) ((Button) event.getSource()).getScene().getWindow();  
Utility.rememberPosition(stage);  
stage.setScene(new Scene(root));  
Utility.centerScreen(stage);  
stage.setTitle("Appointment Scheduler");  
stage.show();
```

Just a heads up i played around with the problem a little more and was able to get it to display correctly without having to disable and re-enable to plugin.

What i ended up doing was I selected everything in the code block and cut it to clipboard leaving the just the java block and line settings like this:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/46eb5d0d-17f8-4311-8a5c-78b05cd9db38)
Then I switch back to reading mode so the code block looks like this:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/de1a2cd2-c3de-49be-92a9-e552a1c7330c)
Then I switch back into edit mode and replace the code i had cut to clipboard, and when I switch back to reading mode it looks the way it should!
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/23444760/5dc9df09-d7c6-49b4-8a30-013e430ecd1a)

So the issue seems to have something to do with refreshing the view maybe? Hope this helps.
